### PR TITLE
Update JEDI hashes in sorc

### LINKED ISSUE #1117: Update hashes for JEDI submodules
The hashes for the core JEDI component submodules in `sorc/` are several weeks old.  This issue is opened to document the updating of the hashes for these JEDI components.

Note:  The updating of `sorc/soca` is documented in GDASApp issue #1093

### PR DESCRIPTION
**Pull request by RussTreadon-NOAA**
This PR updates the hashes for select JEDI repositories in the GDASApp `sorc` directory.  Updating the submodule hashes necessitates corresponding changes elsewhere in GDASApp.   These additional changes include
- soca updates (see GDASApp issue [#1093](https://github.com/NOAA-EMC/GDASApp/issues/1093))
- update names for select radiance bias correction terms
- correcting the path to `satbias2ioda.x` in `test/atm/test_convert_gsi_satbias.sh`

Resolves #1117

### CODE REVIEW

**Comment by RussTreadon-NOAA** [comment]
**NOTE** This PR is opened in draft mode since full functionality of the changes in this PR depends on other PRs
 - OOPS PR [#2645](https://github.com/JCSDA-internal/oops/pull/2645).  Once this PR is merged, the change to `.gitmodules` in the PR will be reverted
 - jcb-gdas PR [#8](https://github.com/NOAA-EMC/jcb-gdas/pull/8)
 - GSI-utils PR [#43](https://github.com/NOAA-EMC/GSI-utils/pull/43) & g-w PR [#2592](https://github.com/NOAA-EMC/global-workflow/pull/2592)

Additionally the amsua_n19 bias correction files in `GDASAPP_TESTDATA` need to be updated to the new bias correction term naming convention.  Once this is done in the authoritative `GDASAPP_TESTDATA` the changes to `modulefiles/GDAS/orion.intel.lua` will be reverted.

**Comment by CoryMartin-NOAA** [comment]
Now that generic diffusion is in JEDI/SABER, @andytangborn will need the latest versions of OOPS and SABER in GDASApp soon to coincide with some workflow changes. Is the OOPS bugfix created by Bo essential to include in this PR (meaning does it affect our current configuration)?

**Comment by RussTreadon-NOAA** [comment]
@CoryMartin-NOAA , OOPS PR [#2645](https://github.com/JCSDA-internal/oops/pull/2645) is required for the JEDI ATM lgetkf to run as it currently does.  Without PR [#2645](https://github.com/JCSDA-internal/oops/pull/2645), lgetkf writes increments for all variables in the input state.  This causes problems for downstream applications.

**Comment by RussTreadon-NOAA** [comment]
@CoryMartin-NOAA :  This PR can be changed to _Ready for review_ along with jcb-gdas PR [#8](https://github.com/NOAA-EMC/jcb-gdas/pull/8) with the caveat that not all existing GDASApp functionality will be available.

**Comment by CoryMartin-NOAA** [comment]
Thanks @RussTreadon-NOAA I think we can wait a few more days to see if that gets merged in

**Comment by CoryMartin-NOAA** [comment]
@RussTreadon-NOAA @guillaumevernieres are you both okay if we just start the process of updating to develop now, with the understanding we'll need to do it again soon for the GETKF changes? @andytangborn would like to move forward with testing of diffusion for the aerosols and that depends on recent OOPS and SABER changes. If so, I can work towards updating the submodules again, and test.

**Comment by guillaumevernieres** [comment]
> @RussTreadon-NOAA @guillaumevernieres are you both okay if we just start the process of updating to develop now, with the understanding we'll need to do it again soon for the GETKF changes? @andytangborn would like to move forward with testing of diffusion for the aerosols and that depends on recent OOPS and SABER changes. If so, I can work towards updating the submodules again, and test.

Yes, sounds good to me @CoryMartin-NOAA

**Comment by RussTreadon-NOAA** [comment]
I'm glad to move this PR forward.  Once it is in I can update g-w PR [#2641](https://github.com/NOAA-EMC/global-workflow/pull/2641).  

We need to come back at a later date and clean up the oops hack.

**Comment by CoryMartin-NOAA** [comment]
oh the hack... yeah we can't merge that in, because people will clone the workflow and it will fail because not everyone who clones the workflow will have access to that...

**Comment by guillaumevernieres** [comment]
> oh the hack... yeah we can't merge that in, because people will clone the workflow and it will fail because not everyone who clones the workflow will have access to that...

we could fork the public oops and add the letkf branch to that?

**Comment by CoryMartin-NOAA** [comment]
Looks like the OOPS PR has been merged, so I think we are ready to update hashes to head of develop for GDASApp.

**Comment by guillaumevernieres** [comment]
> Looks like the OOPS PR has been merged, so I think we are ready to update hashes to head of develop for GDASApp.

should we run the ci first maybe?

**Comment by CoryMartin-NOAA** [comment]
without the hashes updated, @guillaumevernieres ? Or as part of the PR that updates the hashes? This PR will need updated to the latest hashes still, then CI.

**Comment by guillaumevernieres** [comment]
> without the hashes updated, @guillaumevernieres ? Or as part of the PR that updates the hashes? This PR will need updated to the latest hashes still, then CI.

Yes, never mind ... Update + ci. I think we're on the same page.

**Comment by CoryMartin-NOAA** [comment]
@RussTreadon-NOAA is it okay if I merge into your fork, or would you be willing to update the hashes in this PR and then we can run the CI

**Comment by RussTreadon-NOAA** [comment]
Just got out of a meeting.  Let me get to work on this.

**Comment by CoryMartin-NOAA** [comment]
@RussTreadon-NOAA thanks! let me know how I can help

**Comment by CoryMartin-NOAA** [comment]
I'm kicking off the automated testing but I am also going to compile myself.

**Comment by RussTreadon-NOAA** [comment]
> I'm kicking off the automated testing but I am also going to compile myself.

Doing likewise on Orion.

**Comment by CoryMartin-NOAA** [comment]
Looks like a JCB issue on Orion CI, due to an old JCB still in the global-workflow that it checked out. I expect the same issue on Hera.

**Comment by CoryMartin-NOAA** [comment]
After fixing JCB issues, I now get "lapseRate does not exist in ufo::PredictorFactory". @RussTreadon-NOAA is it working for you or did you get the same error?

**Comment by RussTreadon-NOAA** [comment]
Yes, automated CI using g-w `develop` will fail due to g-w using an older snapshot of jcb.  I updated `sorc/jcb` in my working copy on Orion (look in `/work2/noaa/da/rtreadon/git/global-workflow/update_gdasapp`).

47 out of 47 GDASApp ctests pass from my build.  g-w C96C48_ufs_hybatmDA CI has run through the var and lgetkf apps ... provided `sorc/jcb` is updated.

**Comment by CoryMartin-NOAA** [comment]
Hmm, I wonder why mine is not working. It seems like it has an old UFO hash for some reason. Let me double check this after my meeting.

**Comment by CoryMartin-NOAA** [comment]
Ok @guillaumevernieres FYI we can no longer trust the automated CI as for whatever reason it is not updating submodules properly... not intended to hold up this PR (to the contrary) but something we need to sort out ASAP

**Comment by CoryMartin-NOAA** [comment]
Ok, I know what the problem is, the CI does not work on forks, and since this is a branch on Russ's fork, it did not even check out the changes properly.

**Comment by RussTreadon-NOAA** [comment]
> Ok, I know what the problem is, the CI does not work on forks, and since this is a branch on Russ's fork, it did not even check out the changes properly.

Oops, my bad.  Sorry.

**Comment by RussTreadon-NOAA** [comment]
g-w C96C48_ufs_hybatmDA CI has completed the gdas and enkfgdas cycles.  The gfs fcst and downstream jobs remain.  DA jobs successfully ran to completion.

**Comment by CoryMartin-NOAA** [comment]
no problem @RussTreadon-NOAA you've uncovered a flaw in our CI!

I was able to compile your branch manually, that plus your testing, I think is sufficient to get this PR in as it is long overdue.

**Review by CoryMartin-NOAA — APPROVED** [review]
Approved pending successful completion of the forecast jobs in @RussTreadon-NOAA 's test

**Comment by RussTreadon-NOAA** [comment]
g-w C96C48_ufs_hybatmDA CI completed on Orion.  All jobs passed
```
Orion-login-4:/work2/noaa/stmp/rtreadon/EXPDIR/prsub$ rocotostat -d prsub.db -w prsub.xml -c all -s
   CYCLE         STATE           ACTIVATED              DEACTIVATED
202402231800        Done    Jun 05 2024 17:21:32    Jun 05 2024 17:40:28
202402240000        Done    Jun 05 2024 17:21:32    Jun 05 2024 19:40:33
```

**Comment by CoryMartin-NOAA** [comment]
Thanks @RussTreadon-NOAA are we ready to merge this in?

**Comment by RussTreadon-NOAA** [comment]
@CoryMartin-NOAA , yes, let's merge this PR into `develop`.  Other developers want the JEDI repo updates, right?

**Comment by CoryMartin-NOAA** [comment]
Yes, @andytangborn in particular. Thanks @RussTreadon-NOAA I'll merge!

File: utils/soca/gdas_soca_diagb.h
Comment (guillaumevernieres) [inline utils/soca/gdas_soca_diagb.h:303]: Turns out this is using the wrong function space ...

**Pull request merged**
